index.js

```diff
@@ -276,6 +276,143 @@ async function getCircuit(conv) {
   }
 }
 
+
+/**
+ * Set a logged on user's presence
+ */
+app.intent('set.presence', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+  const { presenceType } = conv.contexts.input['setpresence_data'].parameters;
+  const presence = presenceType.toLowerCase();
+  
+  const { untilTime } = conv.contexts.input['setpresence_data'].parameters;
```

Comment:
Why are there two of these? I assume one if for if someone says 'set dnd until x time' and the other is for if someone says 'set dnd for x hours'. trying to combine these somehow might be better.